R/sample_grid.R

```diff
@@ -0,0 +1,109 @@
+##' Take a grid search produced by \code{\link{scan_beta_date}} and 
+##' sample \code{n_sample_pairs} from the parameter grid uses based
+##' on their probaility. For each parameter pair chosen, run particle
+##' filter with \code{num_particles} and sample 1 trajectory
+##' 
+##' @title Sample Grid Scan
+##' 
+##' @param scan_results Output of \code{\link{scan_beta_date}}.
+##' 
+##' @param n_sample_pairs Number of parameter pairs to be sampled. Integer.
+##'   Default = 10.
+##'   
+##' @param n_particles Number of particles. Positive Integer. Default = 100
+##'   
+##' @return List. First element is an array of trajectories, in which
+##'  first dimension is time, second is parameters chosen when sampling from 
+##'  the grid and the third dimension is alist of inputs.
+##' 
+##' @export
+##' @import furrr
+##' @importFrom utils tail
+sample_grid_scan <- function(scan_results, 
+                             n_sample_pairs = 10, 
+                             n_particles = 100) {
+  
+  # checks on args
+  assert_is(scan_results, "sircovid_scan")
+  assert_pos_int(n_sample_pairs)
+  assert_pos_int(n_particles)
+  
+  # grab the pobability matrix
+  prob <- scan_results$renorm_mat_LL
+  nr <- nrow(prob)
+  nc <- ncol(prob)
+  
+  # construct what the x and y dimensions look like
+  x_grid <- matrix(scan_results$x, nrow = nr, ncol = nc)
```

Comment:
Could using `expand.grid()` be used here, which is the same method to generate this as in `sample_grid.R`

Reply:
This is slightly different to how `expand.grid` is used in `grid_search` as trying to get the grid of beta and start values that correspond to the z axis matrix:

```
x_grid
     [,1] [,2]
[1,] 0.10 0.10
[2,] 0.18 0.18
```

Will leave as is if that makes sense

Reply:
Ah right, I misunderstood. Can you just clarify this example output in a comment?

Reply:
yup done